src/main/java/technology/dice/dicefairlink/config/FairlinkConfiguration.java

```diff
@@ -32,11 +32,13 @@
   public static final String DISCOVERY_MODE_PROPERTY_NAME = "discoveryMode";
   public static final String VALIDATE_CONNECTION = "validateConnection";
   public static final String CLUSTER_REGION = "auroraClusterRegion";
+  public static final String FALLBACK_ENDPOINT = "fallbackEndpoint";
   private static final Duration DEFAULT_POLLER_INTERVAL = Duration.ofSeconds(30);
   private static final Duration DEFAULT_TAG_POLL_INTERVAL = Duration.ofMinutes(2);
   private static final String MYSQL = "mysql";
   private final Region auroraClusterRegion;
   private final Optional<String> replicaEndpointTemplate;
+  private final Optional<String> fallbackEndpoint;
```

Comment:
Apparently there are people that argue that having `Optional` as a field is a bad idea. The general agreement is to have a property and wrap the getter in `Optional.ofNullable`.

This is what the immutables do for the generated classes

Reply:
I am 100% of the opposite opinion. I don't see any value in re-boxing an immutable field in an optional each time the getter is called.